Upgrade to 2.6.X may fail when there are imported Solutions

**Component**: upgrade

**What happened**:
When trying to upgrade from 2.5.1 or 2.6.0-dev to the latest 2.6.0-dev build, the process fails with the following errors (200 last lines from the upgrade.log):

```
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for (u'/etc/salt/pki/master', u'bootstrap_master', u'tcp://192.168.1.18:4506'
, u'clear')
[DEBUG   ] Connecting the Minion to the Master URI (for the return server): tcp://192.168.1.18:4506
[DEBUG   ] Trying to connect to: tcp://192.168.1.18:4506
[DEBUG   ] Closing AsyncZeroMQReqChannel instance
[DEBUG   ] LazyLoaded local_cache.get_load
[DEBUG   ] Reading minion list from /var/cache/salt/master/jobs/0c/c73df1c209da320bd10c8fc663dfef0308c088ed253026d338f35aec2678
e4/.minions.p
[DEBUG   ] get_iter_returns for jid 20200923081219190378 sent to set(['bootstrap']) will timeout at 08:12:24.214898
[DEBUG   ] Checking whether jid 20200923081219190378 is still running
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for (u'/etc/salt/pki/master', u'bootstrap_master', u'tcp://192.168.1.18:4506'
, u'clear')
[DEBUG   ] Connecting the Minion to the Master URI (for the return server): tcp://192.168.1.18:4506
[DEBUG   ] Trying to connect to: tcp://192.168.1.18:4506
[DEBUG   ] Closing AsyncZeroMQReqChannel instance
[DEBUG   ] Passing on saltutil error. Key 'u'retcode' missing from client return. This may be an error in the client.
[DEBUG   ] Checking whether jid 20200923081219190378 is still running
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for (u'/etc/salt/pki/master', u'bootstrap_master', u'tcp://192.168.1.18:4506'
, u'clear')
[DEBUG   ] Connecting the Minion to the Master URI (for the return server): tcp://192.168.1.18:4506
[DEBUG   ] Trying to connect to: tcp://192.168.1.18:4506
[DEBUG   ] Closing AsyncZeroMQReqChannel instance
[DEBUG   ] Passing on saltutil error. Key 'u'retcode' missing from client return. This may be an error in the client.
[DEBUG   ] Checking whether jid 20200923081219190378 is still running
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for (u'/etc/salt/pki/master', u'bootstrap_master', u'tcp://192.168.1.18:4506'
, u'clear')
[DEBUG   ] Connecting the Minion to the Master URI (for the return server): tcp://192.168.1.18:4506
[DEBUG   ] Trying to connect to: tcp://192.168.1.18:4506
[DEBUG   ] Closing AsyncZeroMQReqChannel instance
[DEBUG   ] Passing on saltutil error. Key 'u'retcode' missing from client return. This may be an error in the client.
[DEBUG   ] Checking whether jid 20200923081219190378 is still running
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for (u'/etc/salt/pki/master', u'bootstrap_master', u'tcp://192.168.1.18:4506'
, u'clear')
[DEBUG   ] Connecting the Minion to the Master URI (for the return server): tcp://192.168.1.18:4506
[DEBUG   ] Trying to connect to: tcp://192.168.1.18:4506
[DEBUG   ] Closing AsyncZeroMQReqChannel instance
[DEBUG   ] Passing on saltutil error. Key 'u'retcode' missing from client return. This may be an error in the client.
[DEBUG   ] Checking whether jid 20200923081219190378 is still running
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for (u'/etc/salt/pki/master', u'bootstrap_master', u'tcp://192.168.1.18:4506'
, u'clear')
[DEBUG   ] Connecting the Minion to the Master URI (for the return server): tcp://192.168.1.18:4506
[DEBUG   ] Trying to connect to: tcp://192.168.1.18:4506
[DEBUG   ] Closing AsyncZeroMQReqChannel instance
[DEBUG   ] Passing on saltutil error. Key 'u'retcode' missing from client return. This may be an error in the client.
[ERROR   ] Exception occurred in Subscriber while handling stream: Already reading
[DEBUG   ] Closing IPCMessageSubscriber instance
[DEBUG   ] An exception occurred in this state: Already reading
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/salt/state.py", line 1981, in call
    **cdata['kwargs'])
  File "/usr/lib/python2.7/site-packages/salt/loader.py", line 1977, in wrapper
    return f(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/salt/states/saltmod.py", line 331, in state
    cmd_ret = __salt__['saltutil.cmd'](tgt, fun, **cmd_kw)
  File "/usr/lib/python2.7/site-packages/salt/modules/saltutil.py", line 1543, in cmd
    client, tgt, fun, arg, timeout, tgt_type, ret, kwarg, **kwargs)
  File "/usr/lib/python2.7/site-packages/salt/modules/saltutil.py", line 1507, in _exec
    for ret_comp in _cmd(**cmd_kwargs):
  File "/usr/lib/python2.7/site-packages/salt/client/__init__.py", line 851, in cmd_iter
    **kwargs):
  File "/usr/lib/python2.7/site-packages/salt/client/__init__.py", line 1102, in get_iter_returns
    for raw in ret_iter:
  File "/usr/lib/python2.7/site-packages/salt/client/__init__.py", line 1036, in get_returns_no_block
    no_block=True, auto_reconnect=self.auto_reconnect)
  File "/usr/lib/python2.7/site-packages/salt/utils/event.py", line 646, in get_event
    ret = self._get_event(wait, tag, match_func, no_block)
  File "/usr/lib/python2.7/site-packages/salt/utils/event.py", line 544, in _get_event
    raw = self.subscriber.read_sync(timeout=wait)
  File "/usr/lib/python2.7/site-packages/salt/transport/ipc.py", line 693, in read_sync
    return self.io_loop.run_sync(lambda: self._read(timeout))
  File "/usr/lib/python2.7/site-packages/salt/ext/tornado/ioloop.py", line 459, in run_sync
    return future_cell[0].result()
  File "/usr/lib/python2.7/site-packages/salt/ext/tornado/concurrent.py", line 239, in result
    raise_exc_info(self._exc_info)
  File "/usr/lib/python2.7/site-packages/salt/ext/tornado/gen.py", line 1070, in run
    yielded = self.gen.send(value)
  File "/usr/lib/python2.7/site-packages/salt/transport/ipc.py", line 678, in _read
    raise exc_to_raise  # pylint: disable=E0702
AssertionError: Already reading
[ERROR   ] An exception occurred in this state: Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/salt/state.py", line 1981, in call
    **cdata['kwargs'])
  File "/usr/lib/python2.7/site-packages/salt/loader.py", line 1977, in wrapper
    return f(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/salt/states/saltmod.py", line 331, in state
    cmd_ret = __salt__['saltutil.cmd'](tgt, fun, **cmd_kw)
  File "/usr/lib/python2.7/site-packages/salt/modules/saltutil.py", line 1543, in cmd
    client, tgt, fun, arg, timeout, tgt_type, ret, kwarg, **kwargs)
  File "/usr/lib/python2.7/site-packages/salt/modules/saltutil.py", line 1507, in _exec
    for ret_comp in _cmd(**cmd_kwargs):
  File "/usr/lib/python2.7/site-packages/salt/client/__init__.py", line 851, in cmd_iter
    **kwargs):
  File "/usr/lib/python2.7/site-packages/salt/client/__init__.py", line 1102, in get_iter_returns
    for raw in ret_iter:
  File "/usr/lib/python2.7/site-packages/salt/client/__init__.py", line 1036, in get_returns_no_block
    no_block=True, auto_reconnect=self.auto_reconnect)
  File "/usr/lib/python2.7/site-packages/salt/utils/event.py", line 646, in get_event
    ret = self._get_event(wait, tag, match_func, no_block)
  File "/usr/lib/python2.7/site-packages/salt/utils/event.py", line 544, in _get_event
    raw = self.subscriber.read_sync(timeout=wait)
  File "/usr/lib/python2.7/site-packages/salt/transport/ipc.py", line 693, in read_sync
    return self.io_loop.run_sync(lambda: self._read(timeout))
  File "/usr/lib/python2.7/site-packages/salt/ext/tornado/ioloop.py", line 459, in run_sync
    return future_cell[0].result()
  File "/usr/lib/python2.7/site-packages/salt/ext/tornado/concurrent.py", line 239, in result
    raise_exc_info(self._exc_info)
  File "/usr/lib/python2.7/site-packages/salt/ext/tornado/gen.py", line 1070, in run
    yielded = self.gen.send(value)
  File "/usr/lib/python2.7/site-packages/salt/transport/ipc.py", line 678, in _read
    raise exc_to_raise  # pylint: disable=E0702
AssertionError: Already reading

[INFO    ] Completed state [Run the highstate] at time 08:13:08.047263 (duration_in_ms=48955.421)
[INFO    ] Running state [https://127.0.0.1:7443/healthz] at time 08:13:08.047937
[INFO    ] Executing state http.wait_for_successful_query for [https://127.0.0.1:7443/healthz]
[DEBUG   ] Requesting URL https://127.0.0.1:7443/healthz using GET method
[DEBUG   ] Using backend: tornado
[upgrade.log](https://github.com/scality/metalk8s/files/5328138/upgrade.log)

[DEBUG   ] Response Status Code: 200
[INFO    ] Match text "ok" was found. Status 200 was found.
[INFO    ] Completed state [https://127.0.0.1:7443/healthz] at time 08:13:08.185515 (duration_in_ms=137.575)
[DEBUG   ] File /var/cache/salt/master/accumulator/140450248939728 does not exist, no need to cleanup
[DEBUG   ] LazyLoaded local_cache.prep_jid
[DEBUG   ] Sending event: tag = salt/run/20200923080832964507/ret; data = {u'fun_args': [{u'pillar': OrderedDict([(u'orchestrat
e', OrderedDict([(u'node_name', u'bootstrap')]))]), u'mods': [u'metalk8s.orchestrate.deploy_node'], u'saltenv': u'metalk8s-2.6.
0-dev', u'orchestration_jid': u'20200923080625999723'}], u'jid': u'20200923080832964507', u'return': {u'outputter': u'highstate
', u'data': {u'bootstrap_master': {u'http_|-Wait for API server to be available before highstate_|-https://127.0.0.1:7443/healt
hz_|-wait_for_successful_query': {u'comment': u'Match text "ok" was found. Status 200 was found.', u'name': u'https://127.0.0.1
:7443/healthz', u'changes': {}, u'start_time': '08:12:16.163054', u'result': True, u'duration': 71.768, u'__run_num__': 11, u'_
_sls__': u'metalk8s.orchestrate.deploy_node', u'data': {u'body': u'ok', u'status': 200, u'text': u'ok'}, u'__id__': u'Wait for
API server to be available before highstate'}, u'salt_|-Wait minion available_|-metalk8s_saltutil.wait_minions_|-runner': {u'co
mment': u"Runner function 'metalk8s_saltutil.wait_minions' executed.", u'name': u'metalk8s_saltutil.wait_minions', u'__orchestr
ation__': True, u'start_time': '08:11:02.330828', u'result': True, u'duration': 31228.143, u'__run_num__': 7, u'__jid__': u'202
00923081102969534', u'__sls__': u'metalk8s.orchestrate.deploy_node', u'changes': {u'return': {u'comment': u'All minions matchin
g "bootstrap" responded and finished startup state: bootstrap', u'result': True}}, u'__id__': u'Wait minion available'}, u'salt
_|-Set grains_|-Set grains_|-state': {u'comment': u'States ran successfully. No changes made to bootstrap.', u'name': u'Set gra
ins', u'start_time': '08:08:41.747877', u'result': True, u'duration': 22038.113, u'__run_num__': 1, u'__jid__': u'2020092308084
1886169', u'__sls__': u'metalk8s.orchestrate.deploy_node', u'changes': {}, u'__id__': u'Set grains'}, u'metalk8s_cordon_|-Cordo
n the node_|-bootstrap_|-node_cordoned': {'comment': 'The node bootstrap is already cordoned', 'name': u'bootstrap', u'start_ti
me': '08:09:05.144521', 'result': True, u'duration': 130.576, u'__run_num__': 3, u'__sls__': u'metalk8s.orchestrate.deploy_node
', 'changes': {}, u'__id__': u'Cordon the node'}, u'salt_|-Check pillar before salt-minion configuration_|-metalk8s.check_pilla
r_keys_|-function': {u'comment': u'Function ran successfully. Function metalk8s.check_pillar_keys ran on bootstrap.', u'name':
u'metalk8s.check_pillar_keys', u'start_time': '08:09:07.128206', u'result': True, u'duration': 2778.628, u'__run_num__': 5, u'_
_jid__': u'20200923080907271741', u'__sls__': u'metalk8s.orchestrate.deploy_node', u'changes': {u'ret': {u'bootstrap': True}, u
'out': u'highstate'}, u'__id__': u'Check pillar before salt-minion configuration'}, u'salt_|-Sync module on the node_|-saltutil
.sync_all_|-function': {u'comment': u'Function ran successfully. Function saltutil.sync_all ran on bootstrap.', u'name': u'salt
util.sync_all', u'start_time': '08:08:38.827185', u'result': True, u'duration': 2919.26, u'__run_num__': 0, u'__jid__': u'20200
923080838937811', u'__sls__': u'metalk8s.orchestrate.deploy_node', u'changes': {u'ret': {u'bootstrap': {u'serializers': [], u'c
louds': [], u'engines': [], u'grains': [], u'executors': [], u'beacons': [], u'utils': [], u'returners': [], u'modules': [], u'
renderers': [], u'log_handlers': [], u'thorium': [], u'states': [], u'matchers': [], u'sdb': [], u'proxymodules': [], u'output'
: []}}, u'out': u'highstate'}, u'__id__': u'Sync module on the node'}, u'salt_|-Reconfigure salt-minion_|-Reconfigure salt-mini
on_|-state': {u'comment': u'States ran successfully. Updating bootstrap.', u'name': u'Reconfigure salt-minion', u'start_time':
'08:09:09.909985', u'result': True, u'duration': 112394.606, u'__run_num__': 6, u'__jid__': u'20200923080910120271', u'__sls__'
: u'metalk8s.orchestrate.deploy_node', u'changes': {u'ret': {u'bootstrap': {u'pkg_|-Install yum-utils_|-yum-utils_|-installed':
 {u'comment': u'All specified packages are already installed', u'name': u'yum-utils', u'start_time': u'08:09:16.665451', u'resu
lt': True, u'duration': 1422.992, u'__run_num__': 1, u'__sls__': u'metalk8s.repo.redhat', u'changes': {}, u'__id__': u'Install
yum-utils'}, u'ini_|-Ensure yum versionlock plugin is enabled_|-/etc/yum/pluginconf.d/versionlock.conf_|-options_present': {u'c
omment': u'No anomaly detected', u'name': u'/etc/yum/pluginconf.d/versionlock.conf', u'start_time': u'08:09:18.145922', u'resul
t': True, u'duration': 8.53, u'__run_num__': 4, u'__sls__': u'metalk8s.repo.redhat', u'changes': {}, u'__id__': u'Ensure yum ve
rsionlock plugin is enabled'}, u'ini_|-Ensure yum plugins are enabled_|-/etc/yum.conf_|-options_present': {u'comment': u'No ano
maly detected', u'name': u'/etc/yum.conf', u'start_time': u'08:09:18.133333', u'result': True, u'duration': 10.765, u'__run_num
__': 3, u'__sls__': u'metalk8s.repo.redhat', u'changes': {}, u'__id__': u'Ensure yum plugins are enabled'}, u'file_|-Configure
salt minion_|-/etc/salt/minion.d/99-metalk8s.conf_|-managed': {u'comment': u'File /etc/salt/minion.d/99-metalk8s.conf updated',
 u'name': u'/etc/salt/minion.d/99-metalk8s.conf', u'start_time': u'08:09:23.153376', u'result': True, u'duration': 184.975, u'_
_run_num__': 14, u'__sls__': u'metalk8s.salt.minion.configured', u'changes': {u'diff': u'--- \n+++ \n@@ -7,7 +7,7 @@\n use_supe
rseded:\n   - module.run\n \n-log_level_logfile: info\n+log_level_logfile: debug\n \n saltenv: metalk8s-2.6.0-dev\n \n'}, u'__i
d__': u'Configure salt minion'}, u'pkg_|-Install yum-plugin-versionlock_|-yum-plugin-versionlock_|-installed': {u'comment': u'A
ll specified packages are already installed', u'name': u'yum-plugin-versionlock', u'start_time': u'08:09:18.089226', u'result':
 True, u'duration': 42.249, u'__run_num__': 2, u'__sls__': u'metalk8s.repo.redhat', u'changes': {}, u'__id__': u'Install yum-pl
ugin-versionlock'}, u'test_|-Wait until salt-minion restarted_|-Wait until salt-minion restarted_|-configurable_test_state': {u
'comment': u"Wait a bit for 'salt-minion' to restart", u'name': u'Wait until salt-minion restarted', u'start_time': u'08:09:23.
424097', u'result': True, u'duration': 10013.565, u'__run_num__': 17, u'__sls__': u'metalk8s.salt.minion.running', u'changes':
{}, u'__id__': u'Wait until salt-minion restarted'}, u'pkgrepo_|-Configure metalk8s-saltstack repository_|-metalk8s-saltstack_|
-managed': {u'comment': u"Package repo 'metalk8s-saltstack' already configured", u'name': u'metalk8s-saltstack', u'start_time':
 u'08:09:18.399685', u'result': True, u'duration': 94.692, u'__run_num__': 8, u'__sls__': u'metalk8s.repo.redhat', u'changes':
{}, u'__id__': u'Configure metalk8s-saltstack repository'}, u'cmd_|-Refresh yum cache_|-yum clean all_|-run': {u'comment': u'St
ate was not run because none of the onchanges reqs changed', u'__state_ran__': False, u'start_time': u'08:09:18.499794', u'resu
lt': True, u'duration': 0.051, u'__run_num__': 9, u'__sls__': u'metalk8s.repo.redhat', u'changes': {}}, u'file_|-Set metalk8s_o
smajorrelease in yum vars_|-/etc/yum/vars/metalk8s_osmajorrelease_|-managed': {u'comment': u'File /etc/yum/vars/metalk8s_osmajo
rrelease is in the correct state', u'name': u'/etc/yum/vars/metalk8s_osmajorrelease', u'start_time': u'08:09:14.095574', u'resu
lt': True, u'duration': 34.73, u'__run_num__': 0, u'__sls__': u'metalk8s.repo.redhat', u'changes': {}, u'__id__': u'Set metalk8
s_osmajorrelease in yum vars'}, u'module_|-Refresh yum cache_|-Refresh yum cache_|-run': {u'comment': u'State was not run becau
se none of the onchanges reqs changed', u'__state_ran__': False, u'start_time': u'08:09:18.502440', u'result': True, u'duration
': 0.048, u'__run_num__': 10, u'__sls__': u'metalk8s.repo.redhat', u'changes': {}}, u'metalk8s_package_manager_|-Install salt-m
inion_|-salt-minion_|-installed': {u'comment': u'All specified packages are already installed and are at the desired version.\n
Package salt-minion is already set to be held..\nPackage salt is already set to be held.', u'name': u'salt', u'start_time': u'0
8:09:18.518660', u'result': True, u'duration': 4634.185, u'__run_num__': 13, u'__sls__': u'metalk8s.salt.minion.installed', u'c
hanges': {}, u'__id__': u'Install salt-minion'}, u'module_|-Check packages availability_|-Check packages availability_|-wait':
{u'comment': u'', u'name': u'Check packages availability', u'start_time': u'08:09:18.503586', u'result': True, u'duration': 2.1
86, u'__run_num__': 11, u'__sls__': u'metalk8s.repo.redhat', u'changes': {}, u'__id__': u'Check packages availability'}, u'cmd_
|-Restart salt-minion_|-salt-call --local service.restart salt-minion > /dev/null_|-wait': {u'comment': u'Command "salt-call --
local service.restart salt-minion > /dev/null" run', u'name': u'salt-call --local service.restart salt-minion > /dev/null', u's
tart_time': u'08:09:23.341920', u'result': True, u'duration': 78.243, u'__run_num__': 16, u'__sls__': u'metalk8s.salt.minion.ru
nning', u'changes': {u'pid': 21250, u'retcode': None, u'stderr': u'', u'stdout': u''}, u'__id__': u'Restart salt-minion'}, u'te
st_|-Ensure salt-minion running_|-Ensure salt-minion running_|-configurable_test_state': {u'comment': u"Ran 'test.ping'", u'nam
e': u'Ensure salt-minion running', u'start_time': u'08:11:01.566767', u'result': True, u'duration': 37.913, u'__run_num__': 19,
 u'__sls__': u'metalk8s.salt.minion.running', u'changes': {}, u'__id__': u'Ensure salt-minion running'}, u'pkgrepo_|-Configure
metalk8s-scality repository_|-metalk8s-scality_|-managed': {u'comment': u"Package repo 'metalk8s-scality' already configured",
u'name': u'metalk8s-scality', u'start_time': u'08:09:18.157026', u'result': True, u'duration': 77.307, u'__run_num__': 5, u'__s
ls__': u'metalk8s.repo.redhat', u'changes': {}, u'__id__': u'Configure metalk8s-scality repository'}, u'pkgrepo_|-Configure met
alk8s-epel repository_|-metalk8s-epel_|-managed': {u'comment': u"Package repo 'metalk8s-epel' already configured", u'name': u'm
etalk8s-epel', u'start_time': u'08:09:18.317029', u'result': True, u'duration': 81.393, u'__run_num__': 7, u'__sls__': u'metalk
8s.repo.redhat', u'changes': {}, u'__id__': u'Configure metalk8s-epel repository'}, u'pkgrepo_|-Configure metalk8s-kubernetes r
epository_|-metalk8s-kubernetes_|-managed': {u'comment': u"Package repo 'metalk8s-kubernetes' already configured", u'name': u'm
etalk8s-kubernetes', u'start_time': u'08:09:18.235142', u'result': True, u'duration': 80.842, u'__run_num__': 6, u'__sls__': u'
metalk8s.repo.redhat', u'changes': {}, u'__id__': u'Configure metalk8s-kubernetes repository'}, u'service_|-Ensure salt-minion
running_|-salt-minion_|-running': {u'comment': u'Service salt-minion is already enabled, and is running', u'name': u'salt-minio
n', u'start_time': u'08:09:33.443125', u'__id__': u'Ensure salt-minion running', u'duration': 88100.797, u'__run_num__': 18, u'
__sls__': u'metalk8s.salt.minion.running', u'changes': {u'salt-minion': True}, u'result': True}, u'test_|-Repositories configur
ed_|-Repositories configured_|-succeed_without_changes': {u'comment': u'Success!', u'name': u'Repositories configured', u'start
_time': u'08:09:18.509995', u'result': True, u'duration': 1.62, u'__run_num__': 12, u'__sls__': u'metalk8s.repo.redhat', u'chan
ges': {}, u'__id__': u'Repositories configured'}}}, u'out': u'highstate'}, u'__id__': u'Reconfigure salt-minion'}, u'salt_|-Che
ck pillar before etcd deployment_|-metalk8s.check_pillar_keys_|-function': {u'comment': u'Function ran successfully. Function m
etalk8s.check_pillar_keys ran on bootstrap.', u'name': u'metalk8s.check_pillar_keys', u'start_time': '08:11:33.561608', u'resul
t': True, u'duration': 3651.908, u'__run_num__': 8, u'__jid__': u'20200923081133723730', u'__sls__': u'metalk8s.orchestrate.dep
loy_node', u'changes': {u'ret': {u'bootstrap': True}, u'out': u'highstate'}, u'__id__': u'Check pillar before etcd deployment'}
, u'salt_|-Kill kube-controller-manager on all master nodes_|-ps.pkill_|-function': {u'comment': u'One or more requisite failed
: metalk8s.orchestrate.deploy_node.Run the highstate', u'start_time': '08:13:08.194312', u'result': False, u'duration': 0.032,
u'__run_num__': 16, u'__sls__': u'metalk8s.orchestrate.deploy_node', u'changes': {}}, u'http_|-Wait for API server to be availa
ble_|-https://127.0.0.1:7443/healthz_|-wait_for_successful_query': {u'comment': u'Match text "ok" was found. Status 200 was fou
nd.', u'name': u'https://127.0.0.1:7443/healthz', u'changes': {}, u'start_time': '08:13:08.047940', u'result': True, u'duration
': 137.575, u'__run_num__': 14, u'__sls__': u'metalk8s.orchestrate.deploy_node', u'data': {u'body': u'ok', u'status': 200, u'te
xt': u'ok'}, u'__id__': u'Wait for API server to be available'}, u'salt_|-Check pillar before highstate_|-metalk8s.check_pillar
_keys_|-function': {u'comment': u'Function ran successfully. Function metalk8s.check_pillar_keys ran on bootstrap.', u'name': u
'metalk8s.check_pillar_keys', u'start_time': '08:12:16.240886', u'result': True, u'duration': 2846.023, u'__run_num__': 12, u'_
_jid__': u'20200923081216519996', u'__sls__': u'metalk8s.orchestrate.deploy_node', u'changes': {u'ret': {u'bootstrap': True}, u
'out': u'highstate'}, u'__id__': u'Check pillar before highstate'}, u'metalk8s_cordon_|-Uncordon the node_|-bootstrap_|-node_un
cordoned': {u'comment': u'One or more requisite failed: metalk8s.orchestrate.deploy_node.Run the highstate', u'start_time': '08
:13:08.193078', u'result': False, u'duration': 0.339, u'__run_num__': 15, u'__sls__': u'metalk8s.orchestrate.deploy_node', u'ch
anges': {}}, u'salt_|-Run the highstate_|-Run the highstate_|-state': {u'comment': u'An exception occurred in this state: Trace
back (most recent call last):\n  File "/usr/lib/python2.7/site-packages/salt/state.py", line 1981, in call\n    **cdata[\'kwarg
s\'])\n  File "/usr/lib/python2.7/site-packages/salt/loader.py", line 1977, in wrapper\n    return f(*args, **kwargs)\n  File "
/usr/lib/python2.7/site-packages/salt/states/saltmod.py", line 331, in state\n    cmd_ret = __salt__[\'saltutil.cmd\'](tgt, fun
, **cmd_kw)\n  File "/usr/lib/python2.7/site-packages/salt/modules/saltutil.py", line 1543, in cmd\n    client, tgt, fun, arg,
timeout, tgt_type, ret, kwarg, **kwargs)\n  File "/usr/lib/python2.7/site-packages/salt/modules/saltutil.py", line 1507, in _ex
ec\n    for ret_comp in _cmd(**cmd_kwargs):\n  File "/usr/lib/python2.7/site-packages/salt/client/__init__.py", line 851, in cm
d_iter\n    **kwargs):\n  File "/usr/lib/python2.7/site-packages/salt/client/__init__.py", line 1102, in get_iter_returns\n
for raw in ret_iter:\n  File "/usr/lib/python2.7/site-packages/salt/client/__init__.py", line 1036, in get_returns_no_block\n
  no_block=True, auto_reconnect=self.auto_reconnect)\n  File "/usr/lib/python2.7/site-packages/salt/utils/event.py", line 646,
in get_event\n    ret = self._get_event(wait, tag, match_func, no_block)\n  File "/usr/lib/python2.7/site-packages/salt/utils/e
vent.py", line 544, in _get_event\n    raw = self.subscriber.read_sync(timeout=wait)\n  File "/usr/lib/python2.7/site-packages/
salt/transport/ipc.py", line 693, in read_sync\n    return self.io_loop.run_sync(lambda: self._read(timeout))\n  File "/usr/lib
/python2.7/site-packages/salt/ext/tornado/ioloop.py", line 459, in run_sync\n    return future_cell[0].result()\n  File "/usr/l
ib/python2.7/site-packages/salt/ext/tornado/concurrent.py", line 239, in result\n    raise_exc_info(self._exc_info)\n  File "/u
sr/lib/python2.7/site-packages/salt/ext/tornado/gen.py", line 1070, in run\n    yielded = self.gen.send(value)\n  File "/usr/li
b/python2.7/site-packages/salt/transport/ipc.py", line 678, in _read\n    raise exc_to_raise  # pylint: disable=E0702\nAssertio
nError: Already reading\n', u'name': u'Run the highstate', u'start_time': '08:12:19.091842', u'result': False, u'duration': 489
55.421, u'__run_num__': 13, u'__sls__': u'metalk8s.orchestrate.deploy_node', u'changes': {}, u'__id__': u'Run the highstate'},
u'salt_|-Refresh the mine_|-mine.update_|-function': {u'comment': u'Function ran successfully. Function mine.update ran on work
er-1, worker-0, worker-2, master-1, master-0, bootstrap.', u'name': u'mine.update', u'start_time': '08:09:03.787894', u'result'
: True, u'duration': 1352.848, u'__run_num__': 2, u'__jid__': u'20200923080903996341', u'__sls__': u'metalk8s.orchestrate.deplo
y_node', u'changes': {u'ret': {u'worker-1': True, u'worker-0': True, u'worker-2': True, u'master-1': True, u'master-0': True, u
'bootstrap': True}, u'out': u'highstate'}, u'__id__': u'Refresh the mine'}, u'salt_|-Register the node into etcd cluster_|-stat
e.orchestrate_|-runner': {u'comment': u"Runner function 'state.orchestrate' executed.", u'name': u'state.orchestrate', u'__orch
estration__': True, u'start_time': '08:12:10.683322', u'result': True, u'duration': 5475.512, u'__run_num__': 10, u'__jid__': u
'20200923081210797473', u'__sls__': u'metalk8s.orchestrate.deploy_node', u'changes': {u'return': {u'outputter': u'highstate', u
'data': {u'bootstrap_master': {u'metalk8s_|-Check etcd cluster health_|-Check etcd cluster health_|-module_run': {u'comment': u
'metalk8s_etcd.check_etcd_health: cluster is healthy', u'name': [u'metalk8s_etcd.check_etcd_health'], u'start_time': '08:12:15.
488126', u'result': True, u'duration': 572.897, u'__run_num__': 1, u'__sls__': u'metalk8s.orchestrate.register_etcd', u'changes
': {u'metalk8s_etcd.check_etcd_health': 'cluster is healthy'}, u'__id__': u'Check etcd cluster health'}, u'metalk8s_etcd_|-Regi
ster host as part of etcd cluster_|-bootstrap_|-member_present': {'comment': 'Peer URLs: https://192.168.1.18:2380 already exis
ts', 'name': u'bootstrap', u'start_time': '08:12:15.045978', 'result': True, u'duration': 431.931, u'__run_num__': 0, u'__sls__
': u'metalk8s.orchestrate.register_etcd', 'changes': {}, u'__id__': u'Register host as part of etcd cluster'}}}, u'retcode': 0}
}, u'__id__': u'Register the node into etcd cluster'}, u'salt_|-Install etcd node_|-Install etcd node_|-state': {u'comment': u'
States ran successfully. No changes made to bootstrap.', u'name': u'Install etcd node', u'start_time': '08:11:37.215979', u'res
ult': True, u'duration': 33461.708, u'__run_num__': 9, u'__jid__': u'20200923081137304490', u'__sls__': u'metalk8s.orchestrate.
deploy_node', u'changes': {}, u'__id__': u'Install etcd node'}, u'metalk8s_drain_|-Drain the node_|-bootstrap_|-node_drained':
{'comment': 'Eviction complete.', 'name': u'bootstrap', u'start_time': '08:09:05.278998', 'result': True, u'duration': 1848.295
, u'__run_num__': 4, u'__sls__': u'metalk8s.orchestrate.deploy_node', 'changes': {u'bootstrap': {u'bootstrap': 'drained'}}, u'_
_id__': u'Drain the node'}}}, u'retcode': 1}, u'success': False, u'_stamp': '2020-09-23T08:13:08.212398', u'user': u'UNKNOWN',
u'fun': u'runner.state.orchestrate'}
[DEBUG   ] Stream is closed
[DEBUG   ] Closing IPCMessageClient instance
[DEBUG   ] An exception occurred in this state: Stream is closed
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/salt/state.py", line 1981, in call
    **cdata['kwargs'])
  File "/usr/lib/python2.7/site-packages/salt/loader.py", line 1977, in wrapper
    return f(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/salt/states/saltmod.py", line 751, in runner
    **kwargs)
  File "/usr/lib/python2.7/site-packages/salt/modules/saltutil.py", line 1668, in runner
    full_return=full_return)
  File "/usr/lib/python2.7/site-packages/salt/runner.py", line 158, in cmd
    full_return)
  File "/usr/lib/python2.7/site-packages/salt/client/mixins.py", line 241, in cmd
    return self.low(fun, low, print_event=print_event, full_return=full_return)
  File "/usr/lib/python2.7/site-packages/salt/client/mixins.py", line 414, in low
    namespaced_event.fire_event(data, 'ret')
  File "/usr/lib/python2.7/site-packages/salt/utils/event.py", line 931, in fire_event
    self.event.fire_event(data, tagify(tag, base=self.base))
  File "/usr/lib/python2.7/site-packages/salt/utils/event.py", line 745, in fire_event
    self.io_loop.run_sync(lambda: self.pusher.send(msg))
  File "/usr/lib/python2.7/site-packages/salt/ext/tornado/ioloop.py", line 459, in run_sync
    return future_cell[0].result()
  File "/usr/lib/python2.7/site-packages/salt/ext/tornado/concurrent.py", line 239, in result
    raise_exc_info(self._exc_info)
  File "/usr/lib/python2.7/site-packages/salt/ext/tornado/gen.py", line 1064, in run
    yielded = self.gen.throw(*exc_info)
  File "/usr/lib/python2.7/site-packages/salt/transport/ipc.py", line 419, in send
    yield self.stream.write(pack)
  File "/usr/lib/python2.7/site-packages/salt/ext/tornado/gen.py", line 1056, in run
    value = future.result()
  File "/usr/lib/python2.7/site-packages/salt/ext/tornado/concurrent.py", line 239, in result
    raise_exc_info(self._exc_info)
  File "<string>", line 3, in raise_exc_info
StreamClosedError: Stream is closed
[ERROR   ] An exception occurred in this state: Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/salt/state.py", line 1981, in call
    **cdata['kwargs'])
  File "/usr/lib/python2.7/site-packages/salt/loader.py", line 1977, in wrapper
    return f(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/salt/states/saltmod.py", line 751, in runner
    **kwargs)
  File "/usr/lib/python2.7/site-packages/salt/modules/saltutil.py", line 1668, in runner
    full_return=full_return)
  File "/usr/lib/python2.7/site-packages/salt/runner.py", line 158, in cmd
    full_return)
  File "/usr/lib/python2.7/site-packages/salt/client/mixins.py", line 241, in cmd
    return self.low(fun, low, print_event=print_event, full_return=full_return)
  File "/usr/lib/python2.7/site-packages/salt/client/mixins.py", line 414, in low
    namespaced_event.fire_event(data, 'ret')
  File "/usr/lib/python2.7/site-packages/salt/utils/event.py", line 931, in fire_event
    self.event.fire_event(data, tagify(tag, base=self.base))
  File "/usr/lib/python2.7/site-packages/salt/utils/event.py", line 745, in fire_event
    self.io_loop.run_sync(lambda: self.pusher.send(msg))
  File "/usr/lib/python2.7/site-packages/salt/ext/tornado/ioloop.py", line 459, in run_sync
    return future_cell[0].result()
  File "/usr/lib/python2.7/site-packages/salt/ext/tornado/concurrent.py", line 239, in result
    raise_exc_info(self._exc_info)
  File "/usr/lib/python2.7/site-packages/salt/ext/tornado/gen.py", line 1064, in run
    yielded = self.gen.throw(*exc_info)
  File "/usr/lib/python2.7/site-packages/salt/transport/ipc.py", line 419, in send
    yield self.stream.write(pack)
  File "/usr/lib/python2.7/site-packages/salt/ext/tornado/gen.py", line 1056, in run
    value = future.result()
  File "/usr/lib/python2.7/site-packages/salt/ext/tornado/concurrent.py", line 239, in result
    raise_exc_info(self._exc_info)
  File "<string>", line 3, in raise_exc_info
StreamClosedError: Stream is closed

[INFO    ] Completed state [state.orchestrate] at time 08:13:08.246094 (duration_in_ms=275433.158)
[INFO    ] Running state [metalk8s.check_pillar_keys] at time 08:13:08.246518
[INFO    ] Executing state salt.function for [metalk8s.check_pillar_keys]
[DEBUG   ] MasterEvent PUB socket URI: /var/run/salt/master/master_event_pub.ipc
[DEBUG   ] MasterEvent PULL socket URI: /var/run/salt/master/master_event_pull.ipc
time="2020-09-23T08:13:09Z" level=fatal msg="execing command in container failed: command terminated with exit code 137"
<< END >>

This script will now exit
```

**What was expected**:
The upgrade process go to the end without any error. :)

**Steps to reproduce**:
Install any 2.5.1 or 2.6.0-dev and then try to upgrade to a 2.6.0-dev.
It seems that the issue happens only if there is at least a solution imported into the cluster.
Also, this error is not encountered 100% of the time, it looks like it's kind of a flaky, maybe related to resources congestion of the platform, but I easily reproduced it many times.

**Resolution proposal** (optional):

--- a/salt/metalk8s/salt/master/installed.sls
+++ b/salt/metalk8s/salt/master/installed.sls
@@ -8,7 +8,7 @@
 {%- set salt_ip = grains['metalk8s']['control_plane_ip'] -%}
 
 {%- set solution_archives = {} %}
-{%- set solutions_available = pillar.metalk8s.get('solutions', {}).get('available', {}) %}
+{%- set solutions_available = salt.metalk8s_solutions.list_available() %}
 {%- for versions in solutions_available.values() %}
   {%- for version in versions %}
     {%- set version_sanitized = version.id | replace('.', '-') %}

